pkg/gatherers/workloads/gather_workloads_runtime_infos.go

```diff
@@ -0,0 +1,173 @@
+package workloads
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type podWithNodeName struct {
+	podIP    string
+	nodeName string
+}
+
+func gatherWorkloadRuntimeInfos(
+	ctx context.Context,
+	coreClient corev1client.CoreV1Interface,
+) (workloadRuntimes, []error) {
+	start := time.Now()
+
+	runtimePodIPs, err := getInsightsOperatorRuntimePodIPs(ctx, coreClient)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	workloadRuntimeInfos := make(workloadRuntimes)
+	var errors = []error{}
+
+	nodeWorkloadCh := make(chan workloadRuntimesResult)
+	var receiveWg sync.WaitGroup
+	receiveWg.Add(1)
+
+	go func() {
+		defer receiveWg.Done()
+		for infosRes := range nodeWorkloadCh {
+			if infosRes.Error != nil {
+				errors = append(errors, infosRes.Error)
+				return
```

Comment:
I guess `continue` instead of `return` ?

Reply:
🤕 thanks!